File: Ahorn/entities/BoosterRefill.jl
module AnonhelperBoosterRefill

using ..Ahorn, Maple

@mapdef Entity "Anonhelper/BoosterRefill" BoostRefill(x::Integer, y::Integer, oneUse::Bool=false, boostOnEnd::Bool=false)


const placements = Ahorn.PlacementDict(
    "Red Booster Refill (Anonhelper)" => Ahorn.EntityPlacement(
        BoostRefill, 
	"point"
    )
)


function boosterSprite(entity::BoostRefill)
    red = get(entity.data, "boostOnEnd", false)
    
    if red
        return "objects/AnonHelper/boosterRefillOnDash/idle00"

    else
        return "objects/AnonHelper/boosterRefill/idle00"
    end
end

function Ahorn.selection(entity::BoostRefill)
    oneUse = get(entity.data, "oneUse", false)
    x, y = Ahorn.position(entity)
    sprite = boosterSprite(entity)

    return Ahorn.getSpriteRectangle(sprite, x, y)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::BoostRefill, room::Maple.Room)
    sprite = boosterSprite(entity)
    Ahorn.drawSprite(ctx, sprite, 0, 0)
end

end

File: Loenn/entities/booster_refill.lua
local utils = require("utils")

local boostRefill = {}

boostRefill.name = "Anonhelper/BoosterRefill"
boostRefill.depth = -100
boostRefill.placements = {
    {
        name = "boost_refill",
        data = {
            oneUse = false,
            boostOnEnd = false,
        }
    }
}
function boostRefill.texture(room, entity)
    local red = entity.boostOnEnd

    if red then
        return "objects/AnonHelper/boosterRefillOnDash/idle00"

    else
        return "objects/AnonHelper/boosterRefill/idle00"
    end
end

function boostRefill.selection(room, entity)
	local x, y = entity.x or 0, entity.y or 0
    return utils.rectangle(x - 5, y - 4, 10, 9)
end

return boostRefill
